volttron/platform/instance_setup.py

```diff
@@ -391,13 +400,87 @@ def vc_config():
     return config
 
 
+def get_cert_and_key(vhome):
+
+    # Check for existing files first. If present and are valid ask if we are to use that
+
+    master_web_cert = os.path.join(vhome, 'certificates/certs/', MASTER_WEB+".crt")
+    master_web_key = os.path.join(vhome, 'certificates/private/', MASTER_WEB + ".pem")
+    cert_error = True
+
+    if is_file_readable(master_web_cert, False) and is_file_readable(master_web_key, False):
+        try:
+            if certs.Certs.validate_key_pair(master_web_cert, master_web_key):
+                print('\nFollowing certificate and keyfile exists for web access over https: \n{}\n{}'.format(master_web_cert,
+                                                                                                              master_web_key))
+                prompt = '\nDo you want to use these certificates for VC? '
+                if prompt_response(prompt, valid_answers=y_or_n, default='Y') in y:
+                    config_opts['web-ssl-cert'] = master_web_cert
+                    config_opts['web-ssl-key'] = master_web_key
+                    cert_error = False
+                else:
+                    print('\nPlease provide path to cert and key files. '
+                          'This will overwrite existing files: \n{} and {}'.format(master_web_cert, master_web_key))
+            else:
+                print("Existing key pair is not valid. ")
+        except RuntimeError as e:
+            print(e)
+            pass
+
+
+
+    # Either are there no valid existing certs or user decided to overwrite the existing file.
+    # Prompt for new files
+    while cert_error:
+        while True:
+            prompt = 'Enter the SSL certificate public key file:'
+            cert_file = prompt_response(prompt, mandatory=True)
+            if is_file_readable(cert_file):
+                break
+            else:
+                print("Unable to read file {}".format(cert_file))
+        while True:
+            prompt = \
+                'Enter the SSL certificate private key file:'
+            key_file = prompt_response(prompt, mandatory=True)
+            if is_file_readable(key_file):
+                break
+            else:
+                print("Unable to read file {}".format(key_file))
+        try:
+            if certs.Certs.validate_key_pair(cert_file, key_file):
+                cert_error = False
+                shutil.copy(cert_file, master_web_cert)
```

Comment:
I don't think that these should be copied into the masterweb.  This will probably cause a problem with masterweb not being able to connect to rabbitmq as I would doubt that the cert and key have been signed by our root ca.  We should either copy to a new specific location or leave as is with just the filename.

Reply:
we are creating this only in case of zmq + https web-bind-address. So it is not connecting to rmq and won't have any self signed root certs. Or are you thinking of a case where someone switches from RMQ to ZMQ and might want to go back to RMQ with the same vhome? I wouldn't recommend them doing it, but guess in theory it could be done

Reply:
no I am saying that if the user has any cert key we should be able to use it rather than copying it so if someone has a cert and key in ~/.ssh/ and it has a 400 permission on it, then it shouldn't get spread/copied/moved to the masterweb location

Reply:
Valid point. But what if the path is $HOME/filename or $HOME/Downloads/filename or /tmp/filename. I am just thinking it might get accidentally deleted. Should we print some kind of warning or info that we will be simply pointing to it? Or may be we can print the final generated config file at the end?

Reply:
If it accidentally gets deleted then they will get an error the next time volttron is run because they won't have a valid path.  

We can print out a warning saying that if the file isn't chmod 400 then its perhaps in danger just like ssh does if you use an identity file connecting that is group writable.